cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -62,7 +62,7 @@
    *
    * @param id        Info about program
    * @param pid       run id
-   * @param startTime start timestamp
+   * @param startTime start timestamp in seconds, if run id is time-based pass the time from the run id
```

Comment:
Done. It means that if you can derive the time from run id send that time to this function, rather than current time.